openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraFilterSetMethodContract.scala

```diff
@@ -811,4 +811,513 @@ trait LinagoraFilterSetMethodContract {
+                     |	"using": ["com:linagora:params:jmap:filter"],
+                     |	"methodCalls": [
+                     |		["Filter/set", {
+                     |			"accountId": "$generateAccountIdAsString",
+                     |			"ifInState": "-1",
+                     |			"update": {
+                     |				"singleton": [{
+                     |					"id": "1",
+                     |					"name": "My first rule",
+                     |					"condition": {
+                     |						"field": "subject",
+                     |						"comparator": "contains",
+                     |						"value": "question"
+                     |					},
+                     |					"action": {
+                     |						"appendIn": {
+                     |							"mailboxIds": ["$generateMailboxIdForUser"]
+                     |						}
+                     |					}
+                     |				}]
+                     |			}
+                     |		}, "c1"],
+                     |		[
+                     |			"Filter/get",
+                     |			{
+                     |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+                     |				"ids": ["singleton"]
+                     |			},
+                     |			"c2"
+                     |		]
+                     |	]
+                     |}""".stripMargin
+
+    val response = `given`()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |	"sessionState": "${SESSION_STATE.value}",
+         |	"methodResponses": [
+         |		[
+         |			"Filter/set",
+         |			{
+         |				"accountId": "$generateAccountIdAsString",
+         |				"newState": "${INSTANCE.value}",
+         |				"updated": {
+         |					"singleton": {
+         |
+         |					}
+         |				}
+         |			},
+         |			"c1"
+         |		],
+         |		[
+         |			"Filter/get", {
+         |				"accountId": "$generateAccountIdAsString",
+         |				"state": "0",
+         |				"list": [{
+         |					"id": "singleton",
+         |					"rules": [{
+         |						"name": "My first rule",
+         |						"condition": {
+         |							"field": "subject",
+         |							"comparator": "contains",
+         |							"value": "question"
+         |						},
+         |						"action": {
+         |							"appendIn": {
+         |								"mailboxIds": ["$generateMailboxIdForUser"]
+         |							}
+         |						}
+         |					}]
+         |				}],
+         |				"notFound": []
+         |			}, "c2"
+         |		]
+         |	]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def updateWithIfInStateIsInitialWhenAStateIsDefinedShouldSucceed(): Unit = {
```

Comment:
```suggestion
  def updateWithIfInStateIsInitialWhenAStateIsDefinedShouldFail(): Unit = {
```
I mean it's failing at the end right? :)